src/main/java/Duke.java

```diff
@@ -20,207 +20,196 @@ public static void main(String[] args) {
-                    }
-                    System.out.println(border + "Goodbye, then!\n" + border);
-                    LoopEnd = true;
-                    break;
-
-                // Duke lists out all Task names in TaskList when input is "list"
-                case ("list"):
-                    // ERROR: list format is anything other than [ list ]
-                    if (UserScan.nextLine().length()>0) {
-                        System.out.println("\n" + border + "[ERROR]\nUh, list command format is used wrongly.\nCorrect format is as follows:\n" +
-                                "[ list ]\n" + border);
-                        break;
-                    }
-                    if (TaskList.size()==0) {
-                        System.out.println(border + "You don't have anything to do right now!\n" + border);
+                    // loop breaks, ending program if input is "bye"
+                    case ("bye"):
+                        // ERROR: bye format is anything other than [ bye ]
+                        if (UserScan.nextLine().length()>0) {
+                            throw new DukeException("\n" + border + "[ERROR]\nUh, bye command format is used wrongly.\nCorrect format is as follows:\n" +
+                                    "[ bye ]\n" + border);
+                        }
+                        System.out.println(border + "Goodbye, then!\n" + border);
+                        LoopEnd = true;
                         break;
-                    }
-                    String ToPrint = "";
-                    for (int i = 0; i < TaskList.size(); i++) {
-                        ToPrint += ((i + 1 + ". ") + TaskList.get(i).toString() + "\n");
-                    }
-                    System.out.println(border + "Here are your tasks:\n" + ToPrint + border);
-                    break;
 
-                // Duke allows user to mark tasks as done when input is "mark"
-                case ("mark"):
-                    try {
-                        String MarkString = UserScan.nextLine().strip();
-                        // ERROR: mark format is anything other than [ mark <insert integer> ]
-                        if (MarkString.length()==0) {
-                            System.out.println("\n" + border + "[ERROR]\nUh, mark command format is used wrongly.\nCorrect format is as follows:\n" +
-                                    "[ mark <insert INTEGER> ]\n" + border);
+                    // Duke lists out all Task names in TaskList when input is "list"
+                    case ("list"):
+                        // ERROR: list format is anything other than [ list ]
+                        if (UserScan.nextLine().length()>0) {
+                            throw new DukeException("\n" + border + "[ERROR]\nUh, list command format is used wrongly.\nCorrect format is as follows:\n" +
+                                    "[ list ]\n" + border);
+                        }
+                        if (TaskList.size()==0) {
```

Comment:
could this be abstracted out to TaskList.isEmpty for it be more meaningful? just a suggestion!